pumpkin-solver/src/api/solver.rs

```diff
@@ -137,14 +138,26 @@ impl Solver {
 
     /// Logs the statistics currently present in the solver with the provided objective value.
     pub fn log_statistics_with_objective(&self, objective_value: i64) {
-        log_statistic("objective", objective_value);
-        self.log_statistics();
+        let mut writer: Box<dyn Write> = Box::new(stdout());
+        self.write_statistics_with_objective(&mut writer, objective_value);
+    }
+
+    /// Logs the statistics currently present in the solver with the provided objective value.
+    pub fn write_statistics_with_objective(&self, writer: &mut Box<dyn Write>, objective_value: i64) {
```

Comment:
Wouldn't it be possible to use `impl Write` here instead of `Box<dyn Write>`?

Reply:
The issue is that `Propagator::write_statistics` cannot take such a type as it cannot be instantiated at compile time, so therefore a box was necessary everywhere.

Reply:
Good point, since it would be a trait with generics; though I think there are some other functions which could take the opaque `impl Write` type (which do not pass to the trait)?